pkg/operation/common/extensions.go

```diff
@@ -68,26 +70,36 @@ func WaitUntilObjectReadyWithHealthFunction(
 	ctx context.Context,
 	c client.Client,
 	logger *logrus.Entry,
-	healthFunc func(obj runtime.Object) (bool, error),
+	healthFunc func(obj runtime.Object) error,
 	newObjFunc func() runtime.Object,
 	kind string,
 	namespace string,
 	name string,
 	interval time.Duration,
+	severeThreshold time.Duration,
 	timeout time.Duration,
 	postReadyFunc func(runtime.Object) error,
 ) error {
-	var lastObservedError error
+	var (
+		errorWithCode         *gardencorev1beta1helper.ErrorWithCodes
+		lastObservedError     error
```

Comment:
`lastObservedError` is never set 🤔

Reply:
Nice catch, it got accidentally removed in #2336